cobc/parser.y

```diff
@@ -6150,12 +6150,18 @@ record_clause:
 	}
   }
 | RECORD _contains integer TO integer _characters
+  _record_depending /* GCOS extension */
   {
 	check_repeated ("RECORD", SYN_CLAUSE_4, &check_duplicate);
 	if (current_file->organization == COB_ORG_LINE_SEQUENTIAL) {
 		cb_warning (cb_warn_additional, _("RECORD clause ignored for LINE SEQUENTIAL"));
 	} else {
 		set_record_size ($3, $5);
+		if (current_file->record_depending) {
```

Comment:
if there's a duplicate record clause this may be triggered without the clause actually given, to fix this change `_record_depending` to either return its `$3` or (in case of not specified) `NULL`, then adjust this if with using the new token `$7`.
Or: duplicate the complete entry, including _record_depending here, which would allow you to set `current_file->record_depending` based on `cb_verify`.